source/docs/software/basic-programming/java-units.rst

```diff
@@ -0,0 +1,194 @@
+     public static class DriveConstants {
+       /** The distance between the front and rear axles. **/
+       public static final Measure<Distance> kWheelBase = Inches.of(28);
+
+       /** The distance between left and right wheels. **/
+       public static final Measure<Distance> kTrackWidth = Inches.of(26);
+     }
+   }
+
+Because the ``Measure`` objects are Java objects, arithmetic must be done by calling methods on the objects. Note that each operation will return a new ``Measure`` object.
+
+.. code-block:: java
+
+   Measure<Distance> wheelDiameter = Inches.of(3);
+   double gearRatio = 10.48;
+   Measure<Distance> distancePerRotation = wheelDiameter.times(Math.PI).divide(gearRatio);
+
+Arithmetic can also be performed with multiple ``Measure`` objects together:
+
+.. code-block:: java
+
+   Measure<Distance> armLength = Feet.of(3).plus(Inches.of(4.25));
+   Measure<Distance> endEffectorX = armLength.times(Math.cos(getArmAngle().in(Radians));
+   Measure<Distance> endEffectorY = armLength.times(Math.sin(getArmAngle().in(Radians));
+
+Unit conversions can be done by calling ``Measure#in(Unit)``. The Java type system will prevent units from being converted between incompatible types, such as distances to angles. The returned values will be bare ``double`` values without unit information - it is up to you, the programmer, to interpret them correctly!
+
+.. code-block:: java
+
+   Measure<Velocity<Distance>> kMaxVelocity = FeetPerSecond.of(12.5);
+   Measure<Velocity<Velocity<Distance>>> kMaxAcceleration = FeetPerSecond.per(Second).of(22.9);
+
+   // The WPILib math libraries use SI metric units, so we have to convert to meters:
+   TrapezoidProfile.Constraints kDriveConstraints = new TrapezoidProfile.Constraints(
+     maxVelocity.in(MetersPerSecond),
+     maxAcceleration.in(MetersPerSecondSquared)
+   );
+
+Memory Usage and the Garbage Collector
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The roboRIO is a severely memory-constrained runtime environment and the Java garbage collector has to run to keep memory usage to a reasonable level. The garbage collector will have to pause the robot program in order to free unused objects, which manifests as loop overruns and potentially jittery controls. To avoid this issue, allocate as few ``Measure`` objects as possible in areas of code that will run periodically, such as in a Command's ``execute`` or a Subsystem's ``periodic`` method.
+If you still want to use units in hot areas of the code, a special ``MutableMeasure`` class is available. ``MutableMeasure`` allows the internal state of the object to be updated (such as with the results arithmetic operations) to avoid allocating new objects. If the object will be exposed as part of a public API method, have that method return a regular ``Measure`` in its signature to prevent other areas of the code (or users, if it's part of a library) from modifying your internal state.
+
+Revisiting the arm example from above:
+
+.. code-block:: java
```

Comment:
I think it would make more sense to have the discussion of methods before the example

Reply:
Good idea